doc/src/modules/solvers/solveset.rst

```diff
@@ -6,62 +6,84 @@ Solveset
 .. module:: sympy.solvers.solveset
 
 This is the official documentation of the ``solveset`` module in solvers.
-It contains the frequently asked questions about our new module to solve
-equations.
 
 .. note::
 
    For a beginner-friendly guide focused on solving common types of equations,
    refer to :ref:`solving-guide`.
 
-What's wrong with solve():
---------------------------
+How solveset() is different from solve()
+----------------------------------------
+
+SymPy already has a pretty powerful ``solve`` function. But it has different
+goals. For example:
+
+* It has a generalized output interface to handle various types of solutions.
+  It needs to return many types of solutions consistently. It can deal with
+  univariate as well as multivariate equations as well as other kinds
+  of variability due to the system of equations and options passed into it,
+  producing associated outputs. See :ref:`solve_output` for more details.
+
+* The input API has a lot of parameters. It is a facade around
+  more specialized solvers. *This feature of* ``solve`` *could be viewed as analogous
+  to how* ``simplify`` *is a facade around more specialized simplification functions.*
+
+  See :func:`~sympy.solvers.solvers.solve` for more details about the available parameters
+  and output options.
```

Comment:
By `criticizing poor development choices with solve` are you referring to the use of the word `consistently`?

Because I do not hear critique in that section. Now, I certainly do not have the history with the solvers code base that you 2 do.

What I do feel important is calling out the multivariate difference. I have been bit by that multiple times in the last year because I just refused to accept that solveset was truly for univariate equations if it was intended to be a replacement for solve.

I do hear what you are saying, but ...
* it feels like most of that stuff is covered in the next sections
* I have been trying to keep this `differences` section on the light side, adding links to other pages focusing on solve as I went
* the last 2 bullet points were from you - I do realize that quoting you verbatim may not have been what you really wanted - let's refine it. What I do feel strongly about in this blurb is readability. If I do not use bullets the average user will gloss over the points being presented and not read them; missing the full effect. But, with bullets, they can always come back later to review as reference.

What I was hoping for in this first section was half a page of text; brief, concise, to the point. And then get on with the docs for solveset where those kinds of usage examples would appear.

The `Why solveset()?` section has the `sin(x) = 0` example, which is a good one. Is there anything missing that you would like to add?

Should we add the first example (to the same section) to demonstrate the partial solution feature you are highlighting?

With that in mind how can I further simplify the differences section so it is useful to a user?

It feels pretty good to me. It is relatively light and with the new links in place adds to the discoverability.

But please do poke holes in how I see things.

To be honest, I am more concerned about your observation about `criticizing ...` than I am the missing example.

Reply:
The text is using various euphemisms to say that `solve` has a badly designed and inconsistent interface. there is no reason it needs to be so badly designed, just people making poor choices that should have been fixed a long time ago but now it's too difficult because of compatibility:

It should always return a list of dicts:
```python
In [7]: solve(x-1)
Out[7]: [1]

In [8]: solve([x-1])
Out[8]: {x: 1}

In [9]: solve(x**2-1)
Out[9]: [-1, 1]

In [10]: solve([x**2-1])
Out[10]: [{x: -1}, {x: 1}]
```
This isn't a "generalised interface", it is just bad design. It should have been obvious that there would be a need to return multiple solutions (hence a list) and that it can't just be a list of numbers if you want multivariate equations (hence a list of dicts).

The basic format cannot handle inequalities but someone thought it would be good to add that and have it return a completely different type:
```python
In [11]: solve(x**2 < 1)
Out[11]: -1 < x ∧ x < 1

In [12]: solve([x**2-1, x > 0])
Out[12]: x = 1
```
I think most people would want the latter case to return `[{x: 1}]` but if you have inequalities then it returns a Boolean instead of a list of solutions. It was a mistake to add this to `solve` especially since it prevents a more useful interpretation of what inequalities could be used for.

There is a weird handling of equations involving booleans that doesn't really work properly:
```python
In [14]: solve(Eq(Not(x), True), x)
Out[14]: False
```
This should just raise an error instead.

In some situations solve returns an "undetermined coefficients" solution:
```python
In [20]: solve(x**2*y + z, [y, z])
Out[20]: {y: 0, z: 0}

In [21]: solve([x**2*y + z], [y, z])
Out[21]:
⎧   -z ⎫
⎪y: ───⎪
⎨    2 ⎬
⎪   x  ⎪
⎩      ⎭
```
There is a separate function for undetermined coefficients which should be used instead. The nature of the computed solution should not depend on whether there is a single equation or a list of one equation.

None of these are good ideas: it should just always return a list of dicts and it should be for solving equations but perhaps with some support for inequalities that filter the equations.

However none of these things is a reason to use solveset. If you want a consistent return type then you can just always pass a list of equations and use dict=True. The reasons listed here are reasons why sympy developers at the time thought that interface design should be better but they are not reasons for wanting to use mathematical sets instead of a list of dicts.

I agree that multivariate handling should be mentioned since that is an important usability difference. Note that along with solveset are linsolve and nonlinsolve which are supposed to handle multivariate equations but have been misdesigned so that they don't improve on solve in any fundamental way (they don't actually use mathematical sets properly).